GetIntoTeachingApi/Models/TeacherTrainingAdviser/TeacherTrainingAdviserSignUp.cs

```diff
@@ -323,14 +324,31 @@ private void AddQualification(Candidate candidate)
 
         private void AddPastTeachingPosition(Candidate candidate)
         {
-            if (ContainsPastTeachingPosition())
+            // NB:  StageTaughtId is a new parameter and might not be set by older clients.
+            // NB:  If the StageTaughtId==primary, SubjectTaughtId will be null
+            if (candidate.IsReturningToTeaching())
             {
-                candidate.PastTeachingPositions.Add(new CandidatePastTeachingPosition()
+                if (StageTaughtId == (int)CandidatePastTeachingPosition.EducationPhase.Primary ||
+                    SubjectTaughtId == TeachingSubject.PrimaryTeachingSubjectId)
                 {
-                    Id = PastTeachingPositionId,
-                    SubjectTaughtId = SubjectTaughtId,
-                    EducationPhaseId = (int)CandidatePastTeachingPosition.EducationPhase.Secondary,
-                });
+                    candidate.PastTeachingPositions.Add(new CandidatePastTeachingPosition()
+                    {
+                        Id = PastTeachingPositionId,
+                        SubjectTaughtId = TeachingSubject.PrimaryTeachingSubjectId,
+                        EducationPhaseId = (int)CandidatePastTeachingPosition.EducationPhase.Primary,
+                    });
+                }
+                else if ((StageTaughtId == null ||
+                          StageTaughtId == (int)CandidatePastTeachingPosition.EducationPhase.Secondary) &&
+                         SubjectTaughtId != null)
+                {
+                    candidate.PastTeachingPositions.Add(new CandidatePastTeachingPosition()
+                    {
+                        Id = PastTeachingPositionId,
+                        SubjectTaughtId = SubjectTaughtId,
```

Comment:
Is this a Typo ? It looks like self assignment.

Reply:
ah yes, you are right, sorry for that - this line is not required. I'll push an update shortly. Thanks

Reply:
@asatwal actually, on reflection (and tests) - this parameter `SubjectTaughtId` _is_ required. Its part of the initialisation of the CandidatePastTeachingPosition instance:

```
new CandidatePastTeachingPosition()
                    {
                        Id = PastTeachingPositionId,
                        SubjectTaughtId = SubjectTaughtId,
                        EducationPhaseId = (int)CandidatePastTeachingPosition.EducationPhase.Secondary,
                    }
```

which translates to create a new `CandidatePastTeachingPosition` instance setting it's SubjectTaughtId to the request's SubjectTaughtId. I agree its not terribly clear... but that is the c# syntax.